src/expr/src/scalar/func/impls/string.rs

```diff
@@ -568,6 +568,10 @@ impl<'a> EagerUnaryFunc<'a> for CastStringToChar {
         .nullable(input.nullable)
     }
 
+    fn could_error(&self) -> bool {
+        self.fail_on_len && self.length.is_some()
+    }
+
```

Comment:
This seems to be fragile because the function `format_str_trim` (and others) is known to behave like this, but it's not documented to. Maybe we should leave a comment on the function saying that the unary func expects this behavior and things might break should the function change? This assumes people read docs, but I don't think there's a better way to enforce this?

Reply:
Good point, documented!